### PR TITLE
Profile branch

### PR DESCRIPTION
**Pull request by Phenomenalhub**
Created a profile activity with been able to upload a profile picture

### CODE REVIEW

File: app/src/main/java/com/example/studentarena/fragments/FeedFragment.java
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/fragments/FeedFragment.java:79]: delete?

File: app/src/main/java/com/example/studentarena/fragments/ProfileFragment.java
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/fragments/ProfileFragment.java:81]: #13 and this PR i think both have identical code for taking pictures. you should figure out a way to reuse it, maybe by creating a `CameraController.java` file and moving this code to it. so something like
```
// CameraController.java
public class CameraController {
    public static launchCamera(Activity activity, Context context) {
        // blah blah
        intent.resolveActivity(context.getPackageManager()) != null) { /* blah blah */ }
        activity.startActivityForResult(intent, blah);
    }
}
```
```
// ProfileFragment.java
public void onClick(View v) {
    CameraController.launchCamera(this, getContext());
}
```

now, wait until we talk about the git situation before you try to implement this. if we solve it by "rebasing" `Profile_branch` to `main` and separating it from #13, then you'd want to do this in its own PR after merging this and #13. if we solve the git situation by modifying this PR so that it will merge into `New_branch1` instead of `main`, then you can take the suggestion here.

**Review by mhammerly — CHANGES_REQUESTED** [review]
so git is famously hard to use, and i think we're getting hurt by that here

what you intended was to set aside `New_branch1` and create a new `Profile_branch`:
```
`New_branch1` -> o   o <- `Profile_branch`
                 |   |
                 o   o
                  \ /
                   o <- `main`
                   |
                   o
                   |
                   o
```

what actually happened was, your `Profile_branch` branch is branching off of `New_branch1` instead of `main`:
```
o     <- `Profile_branch`
|
o
 \
  o   <- `New_branch1`
  |
  o
   \
    o <- `main`
    |
    o
    |
    o
```

so when you created a PR to merge `Profile_branch` into `main`, it also included everything from `New_branch1` (#13)

using [this link](https://github.com/Phenomenalhub/StudentArena/compare/New_branch1...Profile_branch) i can see the diff between `New_branch1` and `Profile_branch`. i'll leave some inline comments based on that, and tomorrow we can talk about the correct fix for this git mixup

also, there is a "merge conflict". i don't know if GitHub has a good UI to show what exactly happened, but the general concept is:

```
`branch` -> o
            |
            o o <- `main`
             \|
              o
              |
              o
```
in the above situation, `branch` doesn't know about the newest commit from `main`. if they both touch the same file in the same place, we get a "merge conflict" that we have to resolve by figuring out how the two changes can be combined and what the code should look like afterwards.

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java
Comment (mhammerly) [inline app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:87]: i'm still not really understanding why you prefer to keep so many commented-out lines like these. if you want to keep them in case you need to do something similar later, it'd probably be better to take notes in a document somewhere; keeps the code cleaner, and you can cmd+f to search the document whereas with these code comments you have to remember "ahh, i think i used to do that in ComposeFragment.java"

i leave commented-out code sometimes if i expect to uncomment that code in a future commit, like if it uses a feature i haven't implemented yet. otherwise i delete stuff like this, and i think it'd be easier for me to review if you did as well because you have a lot of commented-out code

**Review by mhammerly — COMMENTED** [review]
thanks for addressing the feedback from this PR and from #13! looks great, except you have so much commented-out code that it's making reviews harder

for future reference, if a reviewer requested changes it's best to let them review again after you update your PR before you merge. ideally the reviewer was thorough and left all their feedback in the initial review, but sometimes new things come up after the feedback gets addressed. no big deal for this PR, but keep that in mind in the future

on the other hand, if a reviewer approved the PR but left feedback, you can address the feedback and then merge immediately without waiting on another review

File: app/src/main/res/layout/fragment_compose.xml
Comment (sushant10) [inline app/src/main/res/layout/fragment_compose.xml]: Thoughts on moving this to dimens.xml?

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java
Comment (sushant10) [inline app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:101]: This is a heavy onCreate, see if you can split it up into separate methods

File: app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java
Comment (sushant10) [inline app/src/main/java/com/example/studentarena/fragments/ComposeFragment.java:124]: same nit here on comments

**Review by sushant10 — COMMENTED** [review]
Just took a stab at reviewing this branch!

File: app/src/main/res/layout/fragment_compose.xml
Comment (mhammerly) [reply app/src/main/res/layout/fragment_compose.xml]: not disagreeing but curious as to why - does that make it easier to work with localization/translation down the line or is it just easier to keep all UI text in one place so you don't have to go code spelunking to tweak it?

**Pull request merged**
